homeassistant/components/sensor/zha.py

```diff
@@ -36,7 +36,9 @@
     from zigpy.zcl.clusters.smartenergy import Metering
     from zigpy.zcl.clusters.homeautomation import ElectricalMeasurement
     in_clusters = discovery_info['in_clusters']
-    if RelativeHumidity.cluster_id in in_clusters:
+    if 'sub_component' in discovery_info:
+        sensor = globals()[discovery_info['sub_component']](**discovery_info)
```

Comment:
Whatever it does, it's too much magic.

Reply:
Also, sensor platforms shouldn't be responsible for forwarding. If things need to be dealt by two platforms, it should be routed correctly by the main component

Reply:
Please use `if…else` or create a dict with the classes.

Reply:
Planning on addressing this?